bin/check-include-guards

```diff
@@ -15,21 +21,43 @@ check_guard() {
     fi
 }
 
-for directory in ./exercises/practice/*; do
-    slug=$(basename "${directory}")
-    stub=$(printf "%s" "${slug}" | tr "-" "_")
-    macro=$(printf "%s" "${stub}_H" | tr "[:lower:]" "[:upper:]");
-    guard="#ifndef ${macro}\n#define ${macro}"
-
-    practice="${directory}/${stub}.h"
-    check_guard "$practice" "$guard"
-    (( ERROR_COUNT+=$? ))
+check_exercises () {
+    # Check all exercises in a given directory
+    for directory in "${1}"/*; do
+        slug=$(basename "${directory}")
+        stub=$(printf "%s" "${slug}" | tr "-" "_")
+        macro=$(printf "%s" "${stub}_H" | tr "[:lower:]" "[:upper:]");
+        guard="#ifndef ${macro}\n#define ${macro}"
+
+        practice="${directory}/${stub}.h"
+        check_guard "$practice" "$guard"
+        (( ERROR_COUNT+=$? ))
+
+        example="${directory}/.meta/example.h"
+        check_guard "$example" "$guard"
+        (( ERROR_COUNT+=$? ))
+    done
+}
 
-    example="${directory}/.meta/example.h"
-    check_guard "$example" "$guard"
-    (( ERROR_COUNT+=$? ))
-done
+if [ $# -gt 0 ]
+then
+    echo "Checking include guards format"
+
+    for directory
+    do
+        if [ -d "${directory}" ]
+        then
+            check_exercises "${directory}"
+        else
+            echo "WARNING: Directory not found. Ignoring: ${1}"
```

Comment:
How about making this an error? I can't think of a situation where we wouldn't want to make this obvious.

Reply:
The exercises/concept directory doesn't currently exist. So this would cause the CI to always fail until we get some concept exercises merged

We could maybe not pass that directory in yet, but then at least the first concept exercise will not have CI.
It's kind of a bootstrap problem.

Reply:
My bad, I haven't been following the Concept exercises and didn't realize that none had been merged.